src/overlays/actors/ovl_En_Famos/z_en_famos.c

```diff
@@ -28,133 +62,810 @@ const ActorInit En_Famos_InitVars = {
+    f32 distanceToHome = Actor_XZDistanceToPoint(&this->actor, &this->targetDest);
+
+    this->actor.shape.rot.y = Actor_YawToPoint(&this->actor, &this->targetDest);
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    EnFamos_UpdateBobbingHeight(this);
+    if (this->isCalm) {
+        Math_Vec3f_Copy(&this->calmPos, &this->actor.world.pos);
+    }
+
+    if (EnFamos_IsPlayerSeen(this, globalCtx)) {
+        EnFamos_SetupAlert(this);
+    } else if (distanceToHome < 20.0f) {
+        if (this->pathPoints != NULL) {
+            EnFamos_SetupPathingIdle(this);
+        } else {
+            EnFamos_SetupStillIdle(this);
+        }
+    } else if (distanceToHome < 40.0f) {
+        Math_StepToF(&this->actor.speedXZ, 0.5f, 0.3f);
+    } else {
+        Math_StepToF(&this->actor.speedXZ, 3.0f, 0.3f);
+    }
+}
+
+/**
+ * Famos has spotted the player; delayed by surprise.
+ */
+void EnFamos_SetupAlert(EnFamos* this) {
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    this->stateTimer = 8;
+    this->actor.speedXZ = 0.0f;
+
+    if (this->isCalm == true) {
+        this->isCalm = false;
+        Math_Vec3f_Copy(&this->calmPos, &this->actor.world.pos);
+    }
+
+    this->actionFunc = EnFamos_Alert;
+}
+
+void EnFamos_Alert(EnFamos* this, GlobalContext* globalCtx) {
+    if (ABS_ALT(this->flipRot) > 0x4000) {
+        func_800B9010(&this->actor, NA_SE_EN_FAMOS_FLOAT_REVERSE - SFX_FLAG);
+    } else {
+        func_800B9010(&this->actor, NA_SE_EN_FAMOS_FLOAT - SFX_FLAG);
+    }
+
+    this->stateTimer--;
+    if (this->stateTimer == 0) {
+        this->actor.world.pos.y = this->baseHeight;
+        EnFamos_SetupChase(this);
+    } else {
+        this->actor.world.pos.y = Math_SinS(this->stateTimer * 0x1000) * 30.0f + this->baseHeight;
+    }
+}
+
+/**
+ * Famos has spotted the player; Begin chasing to attack.
+ */
+void EnFamos_SetupChase(EnFamos* this) {
+    this->hoverTimer = 0;
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    this->actionFunc = EnFamos_Chase;
+}
+
+void EnFamos_Chase(EnFamos* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    Vec3f abovePlayerPos;
+    u32 surfaceType;
+
+    EnFamos_UpdateBobbingHeight(this);
+    Math_ScaledStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 0x800);
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    abovePlayerPos.x = player->actor.world.pos.x;
+    abovePlayerPos.y = player->actor.world.pos.y + 100.0f;
+    abovePlayerPos.z = player->actor.world.pos.z;
+    this->actor.world.rot.x = -Actor_PitchToPoint(&this->actor, &abovePlayerPos);
+    Math_StepToF(&this->actor.speedXZ, 6.0f, 0.5f);
+
+    surfaceType = func_800C9B18(&globalCtx->colCtx, this->actor.floorPoly, this->actor.floorBgId);
+    if (this->actor.xzDistToPlayer < 30.0f && this->actor.floorHeight > BGCHECK_Y_MIN && // close enough
+        (surfaceType != 0xC && surfaceType != 0xD)) {
```

Comment:
```suggestion
    if ((this->actor.xzDistToPlayer < 30.0f) && (this->actor.floorHeight > BGCHECK_Y_MIN) && // close enough
        (surfaceType != 0xC && surfaceType != 0xD)) {
```
This can't be 2 ifs, I suppose? Comment not needed IMO.

Reply:
![Screenshot_2022-05-14_14-11-43](https://user-images.githubusercontent.com/42048411/168448323-d26016ef-29db-4460-b66f-a485e1b629f9.png)
Doesn't look like it, because of the following elseif: If condition 1 == TRUE If condition 2 == false leads to falling out of of the if/elseif instead of being caught by the elseif.

Reply:
Yeah, that makes sense.